activesupport/lib/active_support/error_reporter.rb

```diff
@@ -209,13 +209,15 @@ def set_context(...)
     #
     def report(error, handled: true, severity: handled ? :warning : :error, context: {}, source: DEFAULT_SOURCE)
       return if error.instance_variable_defined?(:@__rails_error_reported)
+      raise ArgumentError, "Reported error must be an Exception, got: #{error.inspect}" unless error.is_a?(Exception)
+
       ensure_backtrace(error)
 
       unless SEVERITIES.include?(severity)
         raise ArgumentError, "severity must be one of #{SEVERITIES.map(&:inspect).join(", ")}, got: #{severity.inspect}"
       end
 
-      full_context = ActiveSupport::ExecutionContext.to_h.merge(context)
+      full_context = ActiveSupport::ExecutionContext.to_h.merge(context || {})
```

Comment:
This is too defensive for my taste. It is documented `context` is a hash (at least the default argument), so instead of adding runtime overhead for proper users of this method, lets just make sure the argument are documented properly.

Reply:
I understand the perspective. And I think I would normally agree with you. But since this is already in an error handler I am being defensive on purpose. The overhead is trivial compared to handling an exception. And there is no error handler to handle errors in the error handler. We want whenever possible to report the _intended_ error and I think we should be flexible when doing that as it is friendlier to the user.

But I will also not die on this hill.